src/utils/components/DiskModal/components/DiskSourceSelect/DiskSourceSelect.tsx

```diff
@@ -46,28 +45,14 @@ const DiskSourceSelect: FC<DiskSourceSelectProps> = ({ isTemplate, relevantUploa
                   onChange(val);
                 }}
                 selected={value}
-                selectedLabel={sourceOptions[value]?.label}
+                selectedLabel={optionLabelMapper[value]}
                 toggleProps={{ isDisabled: value === OTHER, isFullWidth: true }}
               >
-                <SelectList>
-                  {Object.entries(sourceOptions).map(([id, { description, label }]) => {
-                    const isDisabled =
-                      (isRunning(vm) && id === SourceTypes.EPHEMERAL) ||
-                      (isCDROMType && id === SourceTypes.BLANK) ||
-                      (isTemplate && id === SourceTypes.UPLOAD);
-                    return (
-                      <SelectOption
-                        data-test-id={`${diskSourceFieldID}-select-${id}`}
-                        description={description}
-                        isDisabled={isDisabled}
-                        key={id}
-                        value={id}
-                      >
-                        {label}
-                      </SelectOption>
-                    );
-                  })}
-                </SelectList>
+                <DiskSourceSelectOptionGroups
+                  isTemplate={isTemplate}
+                  isVMRunning={isRunning(vm)}
+                  optionGroups={optionGroups}
```

Comment:
why passing optionGroups ? i dont ses any other component here using it.
consume it directly at DiskSourceSelectOptionGroups